### ISSUE TITLE
ArrayFire throwing exceptions for some devices and not others

### ISSUE THREAD

**Issue by nlsweet** [arrayfire/arrayfire#2057]
Note: The below example is running on RHEL 7.4.

Using ArrayFire 3.5.1 (installing with the pre-compiled RPM downloaded from your installer archive), the following code crashes on some CUDA devices and not others.

First the code:

```
#include <vector>
#include "arrayfire.h"

int main(int argc, char** argv)
{
    af::info();

    std::vector<double> input = {0, 1, 2, 3, 4};

    af::array afArr1(5, input.data());
    af::array afArr2 = af::constant(1.0, 5, 3, f64);

    for (int j = 1; j < 3; ++j)
    {
        afArr2(af::span, j) = af::pow(afArr1, (double)j);
    }
}
```

And outputs from various devices:
(Note that the binary is compiled on system 1 and then run on the other systems.)

System 1:

ArrayFire v3.5.1 (CUDA, 64-bit Linux, build 0a675e8)
Platform: CUDA Toolkit 8, Driver: 384.111
[0] GeForce GTX 1080 Ti, 11172 MB, CUDA Compute 6.1
afArr2
[5 3 1 1]
    1.0000     0.0000     0.0000
    1.0000     1.0000     1.0000
    1.0000     2.0000     4.0000
    1.0000     3.0000     9.0000
    1.0000     4.0000    16.0000

System 3:

ArrayFire v3.5.1 (CUDA, 64-bit Linux, build 0a675e8)
Platform: CUDA Toolkit 8, Driver: 384.111
[0] GeForce GTX TITAN Z, 6083 MB, CUDA Compute 3.5
-1- GeForce GTX TITAN Z, 6083 MB, CUDA Compute 3.5
terminate called after throwing an instance of 'af::exception'
  what():  ArrayFire Exception (Internal error:998):
In function std::vector<char> cuda::compileToPTX(const char*, std::string)
In file src/backend/cuda/jit.cpp:291
NVRTC Error(7): NVRTC_ERROR unknown

In function void af::array::eval() const
In file src/api/cpp/array.cpp:964

System 4:

ArrayFire v3.5.1 (CUDA, 64-bit Linux, build 0a675e8)
Platform: CUDA Toolkit 8, Driver: 384.111
[0] Tesla K20m, 5062 MB, CUDA Compute 3.5
-1- Tesla K20m, 5062 MB, CUDA Compute 3.5
afArr2
[5 3 1 1]
    1.0000     0.0000     0.0000
    1.0000     1.0000     1.0000
    1.0000     2.0000     4.0000
    1.0000     3.0000     9.0000
    1.0000     4.0000    16.0000 

Note that the code snippet above works fine with AF 3.4.2.

**Comment by Innoket** [comment]
I m facing same error when using FFT.  In my case the Gpu is Nvidia GEFORCE GTX.  But this code is running without any problem on Intel Gpu.

**Comment by 9prady9** [comment]
@nlsweet We have released v3.6 recently. Can you please try your code against v3.6 and let us know if the problem is still present. Thank you!

**Comment by nlsweet** [comment]
Thanks, Pradeep.  I believe this was a problem with the driver, which had a known issue similar to what we were seeing.  We rolled back to the previous driver and ArrayFire version until we have time to test a more stable driver.  I'll close this issue and re-open a new one if the problem persists.
